extensions/core/runtime/src/main/java/org/apache/camel/quarkus/core/runtime/support/FastCamelRuntime.java

```diff
@@ -77,21 +79,17 @@ public void stop() throws Exception {
         doStop();
     }
 
+    @SuppressWarnings("unchecked")
     public void doInit() {
         try {
             this.context = createContext();
 
-            // Configure the camel context using properties in the form:
-            //
-            //     camel.context.${name} = ${value}
-            //
-            RuntimeSupport.bindProperties(properties, context, PFX_CAMEL_CONTEXT);
+            if (properties != null) {
+                PropertyBindingSupport.bindProperties(context, context, new HashMap<String, Object>((Map) properties), PFX_CAMEL_CONTEXT);
```

Comment:
Is there any good reason to allocate a new `HashMap` ? Simply `(Map) props` would compile too.

Reply:
fixed